packages/truffle-decoder/lib/decode/storage.ts

```diff
@@ -39,79 +75,94 @@ export default async function decodeStorageReference(definition: DecodeUtils.Ast
       }
       debug("length %o", length);
 
-      const baseDefinition = DecodeUtils.Definition.baseDefinition(definition);
+      const baseDefinition = definition.baseType || definition.typeName.baseType;
```

Comment:
What happened here? Shouldn't this use the function you just fixed?

Reply:
I didn't fix it -- I removed it.  I didn't see a need for a separate function.  Later when I go through and make it so that we're always assuming `definition` is a type definition and not a variable definition, I'll remove all the duplication.